// user-startup/lib/startup/linux.js
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});
exports.getFile = getFile;
exports.create = create;
exports.remove = remove;
var fs = require('fs');
var cp = require('child_process');
var mkdirp = require('mkdirp');
var untildify = require('untildify');

var dir = untildify('~/.config/autostart');

exports.dir = dir;
function spawn(cmd, args, out) {
  var fd = fs.openSync(out, 'w');
  cp.spawn(cmd, args, {
    stdio: ['ignore', fd, fd],
    detached: true
  }).on('error', console.log).unref();
}

function getFile(name) {
  return '' + dir + '/' + name + '.desktop';
}

function create(name, cmd, args, out) {
  var file = getFile(name);

  var data = ['[Desktop Entry]', 'Type=Application', 'Vestion=1.0', 'Name=${name}', 'Comment=' + name + ' startup script', 'Exec=' + cmd + ' ' + args + ' > ' + out, 'StartupNotify=false', 'Terminal=false'].join('\n');

  mkdirp.sync(dir);
  fs.writeFileSync(file, data);

  spawn(cmd, args, out);
}

function remove(name) {
  var file = getFile(name);
  if (fs.existsSync(file)) fs.unlinkSync(file);
}
