// user-startup/lib/startup/win32.js
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});
exports.getFile = getFile;
exports.create = create;
exports.remove = remove;
var fs = require('fs');
var cp = require('child_process');
var mkdirp = require('mkdirp');
var untildify = require('untildify');

var dir = untildify('~\\AppData\\Roaming\\Microsoft\\Windows\\Start Menu\\Programs\\Startup');

exports.dir = dir;

function getFile(name) {
  return '' + dir + '\\' + name + '.vbs';
}

function create(name, cmd, args, out) {
  var file = getFile(name);

  var escapedCmd = '""' + cmd + '""';
  var escapedArgs = args.map(function (a) {
    return '""' + a + '""';
  }).join(' ');
  var escapedOut = '""' + out + '""';

  var command = '""' + escapedCmd + ' ' + escapedArgs + ' > ' + escapedOut + '""';

  var data = 'CreateObject("Wscript.Shell").Run "cmd /c ' + command + '", 0, true';

  mkdirp.sync(dir);
  fs.writeFileSync(file, data);

  // Spawn vbscript
  cp.spawn('cmd', ['/c', file], {
    stdio: 'ignore',
    detached: true
  }).unref();
}

function remove(name) {
  var file = getFile(name);
  if (fs.existsSync(file)) fs.unlinkSync(file);
}
